File: src/main/java/my/busbookingsystem/Service/PassengerService.java
package my.busbookingsystem.Service;

import my.busbookingsystem.Entity.Passenger;
import my.busbookingsystem.Repository.PassengerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class PassengerService {

    @Autowired
    private PassengerRepository passengerRepository;

    // --- Admin: View All ---
    public List<Passenger> getAllPassengers() {
        return passengerRepository.findAll();
    }

    // --- Admin: Search by Username (Case Insensitive) ---
    public List<Passenger> searchPassengers(String keyword) {
        return passengerRepository.findByUserNameContainingIgnoreCase(keyword);
    }

    // --- Admin: Update Passenger (e.g. if they forgot password or need details changed) ---
    public Passenger updatePassenger(Long id, Passenger updatedInfo) {
        return passengerRepository.findById(id).map(passenger -> {
            passenger.setUserName(updatedInfo.getUserName());
            passenger.setPhoneNo(updatedInfo.getPhoneNo());
            passenger.setAddress(updatedInfo.getAddress());
            // Note: Password usually requires encoding, handled separately
            return passengerRepository.save(passenger);
        }).orElse(null);
    }

    // --- Admin: Delete Passenger ---
    public void deletePassenger(Long id) {
        passengerRepository.deleteById(id);
    }

    //Helper for other services
    public Optional<Passenger> getPassengerById(Long id){
        return passengerRepository.findById(id);
    }

    // Inside PassengerService.java
    public Passenger savePassenger(Passenger passenger) {
        return passengerRepository.save(passenger);
    }

}

File: frontend/src/services/PassengerService.js
import axios from 'axios';

const API_URL = "http://localhost:8080/api/passengers";

class PassengerService {

    // --- 1. Get All Passengers (Admin) ---
    getAllPassengers() {
        return axios.get(`${API_URL}/all`);
    }

    // --- 2. Search Passengers (Admin Search Bar) ---
    searchPassengers(name) {
        return axios.get(`${API_URL}/search`, {
            params: { name: name }
        });
    }

    // --- 3. Update Passenger (Admin) ---
    updatePassenger(passengerId, passengerData) {
        return axios.put(`${API_URL}/update/${passengerId}`, passengerData);
    }

    // --- 4. Delete Passenger (Admin) ---
    deletePassenger(passengerId) {
        return axios.delete(`${API_URL}/delete/${passengerId}`);
    }

    // Inside src/services/PassengerService.js

    // ... existing methods ...

    // --- 5. Register (Sign Up) ---
    register(passengerData) {
        return axios.post(`${API_URL}/register`, passengerData);
    }


    login(credentials) {
        return axios.post(`${API_URL}/login`, credentials);
    }

}

export default new PassengerService();